signer/jws.go

```diff
@@ -188,11 +196,19 @@ func populateProtectedHeaders(protectedHdr *jwsProtectedHeader, signInfo *Signer
 	}
 
 	signInfo.PayloadContentType = protectedHdr.ContentType
-	signInfo.SignedAttributes.SigningTime = protectedHdr.SigningTime.Truncate(time.Second)
+	signInfo.SignedAttributes.ExtendedAttributes = getExtendedAttributes(protectedHdr.ExtendedAttributes, protectedHdr.Critical)
+	signInfo.SigningScheme = protectedHdr.SigningScheme
+	signInfo.SignedAttributes.VerificationPlugin = protectedHdr.VerificationPlugin
+	signInfo.SignedAttributes.VerificationPluginMinVersion = protectedHdr.VerificationPluginMinVersion
 	if protectedHdr.Expiry != nil {
-		signInfo.SignedAttributes.Expiry = protectedHdr.Expiry.Truncate(time.Second)
+		signInfo.SignedAttributes.Expiry = *protectedHdr.Expiry
```

Comment:
nit: No need to use a pointer reference here and below

Reply:
Here we are copying data refereed by `protectedHdr.Expiry` pointer to `signInfo.SignedAttributes.Expiry`. 
`protectedHdr.Expiry` needs to be a pointer because expiry is optional field and shouldn't be present in signature if user haven't specified expiry.